consolidator/consolidator_service.bal

```diff
@@ -92,19 +85,14 @@ isolated function processTopicDeregistration(json payload) returns error? {
     }
 }
 
-isolated function retrieveTopicDeregistration(json payload) returns websubhub:TopicDeregistration|error {
-    string topic = check payload.topic;
-    return {
-        topic: topic
-    };
-}
-
 isolated function processSubscription(json payload) returns error? {
     websubhub:VerifiedSubscription subscription = check payload.cloneWithType(websubhub:VerifiedSubscription);
     string groupName = util:generateGroupName(subscription.hubTopic, subscription.hubCallback);
     lock {
         // add the subscriber if subscription event received
-        subscribersCache[groupName] = subscription.cloneReadOnly();
+         if !subscribersCache.hasKey(groupName) {
+            subscribersCache[groupName] = subscription.cloneReadOnly();
```

Comment:
the key here is suppose to be subscriberId, but calling this groupName and the util method name is confusing